tensorflow_addons/image/filters.py

```diff
@@ -204,3 +200,78 @@ def median_filter2d(
+
+
+def gaussian_filter2d(
+    image: FloatTensorLike,
+    sigma: FloatTensorLike,
+    filter_shape: int,
+    padding: str = "REFLECT",
+    constant_values: TensorLike = 0,
+) -> FloatTensorLike:
+    """
+    This function is responsible for having Gaussian Blur. It takes the image as input, computes a gaussian-kernel
+    which follows normal distribution then convolves the image with the kernel. It is implemented as 2 1D convolutions.
+    Args:
+    image: A tensor of shape
+        (batch_size, height, width, channels)
+        (NHWC), (batch_size, channels, height, width)(NCHW).
+    sigma:A constant of type float64. It is the standard deviation of the normal distribution.
+          The more the sigma, the more the blurring effect.
+          G(x,y)=1/(2*3.14*sigma**2)e^((x**2+y**2)/2sigma**2)
+          In 1D,
+          G(x)=e^(-x**2)/2*sigma**2
+    filter_shape:It is the kernel-size for the Gaussian Kernel.
+          A kernel of size [kernel_shape*kernel_shape] is generated.
+    padding:A string. It takes values in ["REFLECT", "CONSTANT", "SYMMETRIC"].
+    constant_values:A constant to be used for padding in case of CONSTANT padding.
+    Returns:
+        3D or 4D 'Tensor' of same type float64.
+    Raises:
+        Value error if:
+            1). Sigma=0
+            2). passing some string other than ["REFLECT", "CONSTANT", "SYMMETRIC"] in padding.
+    """
+
+    if sigma == 0:
+        raise ValueError("Sigma should not be zero")
+    if padding not in ["REFLECT", "CONSTANT", "SYMMETRIC"]:
+        raise ValueError("Padding should be REFLECT, CONSTANT, OR SYMMETRIC")
+    image = img_utils.to_4D_image(image)
+    channels = tf.shape(image)[3]
+
+    gaussian_filter_x = _get_gaussian_kernel(sigma, filter_shape, channels)
+    gaussian_filter_x = tf.repeat(gaussian_filter_x, channels)
+    gaussian_filter_x = tf.reshape(gaussian_filter_x, [1, filter_shape, channels, 1])
+
+    gaussian_filter_x = tf.cast(gaussian_filter_x, tf.float64)
+    gaussian_filter_y = _get_gaussian_kernel(sigma, filter_shape, channels)
+    gaussian_filter_y = tf.repeat(gaussian_filter_y, channels)
+    gaussian_filter_y = tf.reshape(gaussian_filter_y, [filter_shape, 1, channels, 1])
+
+    gaussian_filter_y = tf.cast(gaussian_filter_y, tf.float64)
+    image = _pad(
+        image,
+        (filter_shape, filter_shape),
+        mode=padding,
+        constant_values=constant_values,
+    )
+    conv_ops_x = tf.nn.depthwise_conv2d(
+        input=image, filter=gaussian_filter_x, strides=(1, 1, 1, 1), padding="VALID"
+    )
+    conv_ops = tf.nn.depthwise_conv2d(
+        input=conv_ops_x,
+        filter=gaussian_filter_y,
+        strides=(1, 1, 1, 1),
+        padding="VALID",
+    )
+    return conv_ops
+
+
+def _get_gaussian_kernel(sigma, filter_shape, channels):
+    "This function creates a kernel of size [filter_shape]"
+    x = tf.range(-filter_shape // 2 + 1, filter_shape // 2 + 1)
+    x = tf.math.square(x, tf.float64)
```

Comment:
Until unless there is an overflow when using`float32`, we should stick to `float32`. If there is any other reason, please let us know

Reply:
Addressed!!!